### ISSUE TITLE
Support np.ndarray for init

### ISSUE THREAD

**Issue by smoh** [stan-dev/pystan#259]
Currently np.ndarray for vector/array parameter in init leads to 

```python
TypeError: Object of type ndarray is not JSON serializable
```

MWE to reproduce the error:
```
import numpy as np
import stan

model_code = '''
data {
    vector[5] x;
}
parameters {
    vector[2] p;
}
model {
    x ~ normal(p[1], p[2]);
}
'''


data = {
        'x':[1,2,3,4,5]
        }
posterior = stan.build(model_code, data=data)
init = [{'p': [0,1]} for i in range(4)]
posterior.sample(init=init)
init_np = [{'p': np.array([0,1])} for i in range(4)]
posterior.sample(init=init_np)
```

To convert them myself, I can do something like:
```python
# python3 does not know how to encode np.ndarray
def convert_types(d):
    for k, v in d.items():
        if isinstance(v, np.ndarray):
            d[k] = v.tolist()
    return d
```
but I would think the functionality is already somewhere in here or httpstan as np.ndarray for data input works fine.

**Comment by riddell-stan** [comment]
Good catch. This one needs the same fix we used in `model.py` for `data`.

**Comment by riddell-stan** [comment]
Closed by #258
